wp-admin/js/widgets/media-widgets.js

```diff
@@ -502,20 +506,86 @@ wp.mediaWidgets = ( function( $ ) {
 		 * @param {wp.media.view.MediaFrame.Select} mediaFrame - Select frame.
 		 * @returns {Object} Props.
 		 */
-		getSelectFrameProps: function getSelectFrameProps( mediaFrame ) {
-			var attachment, props;
+		getModelPropsFromMediaFrame: function getModelPropsFromMediaFrame( mediaFrame ) {
+			var control = this, state, mediaFrameProps;
+
+			state = mediaFrame.state();
+			if ( 'insert' === state.get( 'id' ) ) {
+				mediaFrameProps = _.extend(
+					state.get( 'selection' ).first().toJSON(),
+					mediaFrame.content.get( '.attachments-browser' ).sidebar.get( 'display' ).model.toJSON()
+				);
+				if ( mediaFrameProps.sizes[ mediaFrameProps.size ] ) {
+					mediaFrameProps.url = mediaFrameProps.sizes[ mediaFrameProps.size ].url;
+				}
+			} else if ( 'embed' === state.get( 'id' ) ) {
+				mediaFrameProps = _.extend(
+					state.props.toJSON(),
+					{
```

Comment:
Yes, that would probably make it more intuitive when reading the source. Ultimately unrecognized properties will get dropped when the props get passed through `mapMediaToModelProps`, but definitely the "full" value here is definitely image-specific and it should not be here in the base class.